migrate/live-migration/live-migration-from-postgres.md

```diff
@@ -0,0 +1,321 @@
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --schema-only \
+  --file=dump.sql \
+  --snapshot=$(cat /tmp/pgcopydb/snapshot)
+```
+
+- `--schema-only` is used to dump only the object definitions (schema), not
+  data.
+
+- `--snapshot` is use to specified the synchronized [snapshot][snapshot] when
+  making a dump of the database.
+
+<ExplainPgDumpFlags />
+
+### 3c. Load the roles and schema into the target database
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  --echo-errors \
+  -f roles.sql \
+  -f dump.sql
+```
+
+## 4. Convert hypertables and enable Timescale features
+
+This is the ideal point to convert regular tables into hypertables. In simple
+terms, you'll want to convert the tables that contain time series data. For
+each table that's going to be converted to a Hypertable in the target database,
+run the following command:
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  -c "SELECT create_hypertable('<table name>', '<time column name>')"
+```
+
+A more detailed explanation can be found in the [hypertable
+documentation][hypertable-docs].
+
+Once the table is converted, you can follow the guides to enable more Timescale
+features like [retention][retention-docs], [compression][compression-docs], and
+[data tiering][data-tiering-docs].
+
+This step is optional. Although it's strongly recommended to perform it at this
+point, it can be done after all the data has been migrated, since Hypertables
+can be created from tables with existing data.
+
+The disadvantage of converting after inserting all the data is that the table
+must be locked while the data is being partitioned. This means that no new data
+can be written into the table while the table is being converted. The larger
+the table, the longer it is going to remain locked. Conversely, inserting into
+an already converted Hypertable is a fast operation that partitions the data
+while it's being inserted.
+
+## 5. Migrate initial data from source to target
+
+Use `pgcopydb` and the snapshot ID obtained in the [step 2][step-2]
+(`/tmp/pgcopydb/snapshot`) to copy the data:
+
+```sh
+pgcopydb copy table-data \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --snapshot $(cat /tmp/pgcopydb/snapshot) \
+  --split-tables-larger-than <same-table concurrency size threshold>  \
+  --table-jobs <jobs: usually the number of core in SOURCE>
+```
+
+## 6. Apply the replication changes
+
+With the schema and initial data loaded, you can now apply the buffered and
+live changes emitted by the replication slot:
+
+```sh
+pgcopydb stream sentinel set apply --source $SOURCE
+```
+
+The replication progress can be monitored by querying the [pg_stat_replication
+view][pg-stat-replication-docs]:
+
+```sh
+psql $SOURCE \
+  -f - <<'EOF'
+SELECT replay_lag
+FROM pg_stat_replication
+WHERE application_name='pgcopydb' AND state='streaming'
+EOF
+```
+
+- `replay_lag`: Time elapsed between flushing recent WAL locally and receiving
```

Comment:
What is the unit? Seconds? Milliseconds?

Reply:
The type is interval so it's based on the intervalstyle variable, by default it's pretty printed and depending on how big it's there's more or less information:

https://www.postgresql.org/docs/16/datatype-datetime.html#DATATYPE-INTERVAL-OUTPUT

```
adn@/tmp:adn> select interval '1 second'
╒══════════╕
│ interval │
╞══════════╡
│ 0:00:01  │
╘══════════╛
SELECT 1
Time: 0.004s
adn@/tmp:adn> select interval '1 month'
╒══════════════════╕
│ interval         │
╞══════════════════╡
│ 30 days, 0:00:00 │
╘══════════════════╛
```